### PR TITLE
add document for node qos ensurance

### PR DESCRIPTION
**Pull request by yan234280533**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,19 @@ Two Recommendations are currently supported:
 - **Effective HPARecommend**: Recommend which workloads are suitable for autoscaling and provide optimized configurations such as minReplicas, maxReplicas.
 
 ### QoS Ensurance
+QoS Ensurance gaurantees the stability of the pods running on Kubernetes.
+Users can define PrioirtyClass for the pods and QoSEnsurancePolicy, and then crane detects disruption and ensure the high priority pods not being impacted by resource competition。
```
Comment (mfanjie) [inline README.md:63]: suggest to change to the following statement.
`Kubernetes is capable of starting multiple pods on same node, and as a result, some of the user applications may be impacted when there are resources(e.g. cpu) consumption competition.
To mitigate this, Crane allows users defining PrioirtyClass for the pods and QoSEnsurancePolicy, and then detects disruption and ensure the high priority pods not being impacted by resource competition.`

File: README.md [hunk 2]
```diff
@@ -59,6 +59,19 @@ Two Recommendations are currently supported:
...
+
+Metrics Sources:
+- **NodeLocal**: node resource metrics
+- **ApplicationMetrics**: application metrics
```
Comment (mfanjie) [inline README.md:67]: please remove this or mark it is in future releases

File: README.md (hunk 1 above)
Comment (yan234280533) [reply README.md]: Modified it.

File: README.md (hunk 2 above)
Comment (yan234280533) [reply README.md]: Removed it. Simplicity description first, progressive elaboration in the fureture.

File: docs/tutorials/using-qos-ensurance.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+2. anomaly analyzer: analyze the node triggered anomaly used collected metrics
+3. avoidance action: execute avoidance actions, include disable scheduling, throttle and eviction.
+
+Below is the architecture image.
+<div align="center"><img src="../images/crane-qos-ensurance.png" style="width:900px;" /></div>
+
+The main processes included the following steps:
+1. Metrics collector synchronized policies from kube-apiserver.
+2. Check to see if the policies have been changed. If the policies are changed, the metrics collector updates the collectors.
+3. Metrics collector collects metrics periodically.
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:17]: State collector

File: docs/tutorials/using-qos-ensurance.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+4. Transmit metrics to anomaly analyzer.
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:18]: State collector transmits

File: docs/tutorials/using-qos-ensurance.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+5. Range all rules to analyze the avoidance threshold or the restored threshold reached.
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:19]: Anomaly analyzer ranges

File: docs/tutorials/using-qos-ensurance.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+6. Merged for the analyzed results and notice the avoidance actions.
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:20]: 6. Anomaly analyzer merges for the analyzed results and notice the avoidance actions.

File: docs/tutorials/using-qos-ensurance.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+7. Avoidance action executes actions based on the analyzed results.
+
+# The first user case
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:23]: Disable Scheduling

File: docs/tutorials/using-qos-ensurance.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+We can define an AvoidanceAction and NodeQOSEnsurancePolicy when the node CPU usage triggers the threshold, executes
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:25]: The following AvoidanceAction and NodeQOSEnsurancePolicy can be defined. As a result, when the node CPU usage triggers the threshold, and disable schedule action for the node will be executed

File: docs/tutorials/using-qos-ensurance.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    avoidanceThreshold: 2     # Consider the rule is triggered, when the threshold reached continued so many times
+    restoreThreshold: 2    # Restored the status, when the threshold not reached continued so many times
+    actionName: "disablescheduling" # The name of AvoidanceAction which be associated  
+    strategy: "None"  # Strategy for the action, you can set it "Preview" to not  perform actually
+    metricRule:
+      name: "cpu_total_usage" # The name of metrics
+      value: 4000   # The threshold of metrics
+```
+
+Users can verify that disable scheduling example according to the video below.
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:66]: Please check the video to learn more about the scheduling disable actions.

File: docs/tutorials/using-qos-ensurance.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+[![disable scheduling example video](../images/disablescheduling-example.png)](https://youtu.be/87bnz5LasbI "disable scheduling")
+
+
+# Other user cases
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:71]: Throttle

File: docs/tutorials/using-qos-ensurance.md [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+We can define an AvoidanceAction and NodeQOSEnsurancePolicy when the node CPU usage triggers the threshold, executes
```
Comment (mfanjie) [inline docs/tutorials/using-qos-ensurance.md:72]: similar as above

**Pull request merged**